pySDC/projects/PinTSimE/switch_estimator.py

```diff
@@ -156,13 +149,13 @@ def post_step_processing(self, controller, S):
 
         L = S.levels[0]
 
-        if self.switch_detected_step:
-            if L.prob.params.set_switch[self.count_switches] and L.time + L.dt >= self.t_switch:
-                self.count_switches += 1
-                self.t_switch = None
-                self.switch_detected_step = False
+        if self.params.switch_detected_step:
+            if L.time + L.dt >= self.params.t_switch:
+                self.params.t_switch = None
```

Comment:
There is a function you can overload [here](https://github.com/Parallel-in-Time/pySDC/blob/41a1113dc37edd29c636fb2182ee92bbda02fdf2/pySDC/core/ConvergenceController.py#L150) with resetting of status variables. It doesn't change the functionality I would hope, but I find it cleans the code up to write everything in a function that does exactly what it is supposed to and nothing else. Although this is personal preference, as becomes evident when looking though the controller module for instance. ;)